massa-pool-worker/src/denunciation_pool.rs

```diff
@@ -81,10 +81,11 @@ impl DenunciationPool {
             now,
         );
 
-        if slot.period
-            < self.last_cs_final_periods[slot.thread as usize]
-                .saturating_sub(self.config.denunciation_expire_periods)
-        {
+        if Denunciation::is_expired(
+            &slot.period,
+            &self.last_cs_final_periods[slot.thread as usize],
```

Comment:
Warning: denunciations are not threaded !

Let's update both the pruning and `add_denunciation_precursor` to use `self.last_cs_final_periods.iter().min()` for this check

Reply:
done